config/validator.go

```diff
@@ -0,0 +1,75 @@
+// Copyright © 2022 Meroxa, Inc. & Yalantis
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"fmt"
+	"sync"
+
+	v "github.com/go-playground/validator/v10"
+	"go.uber.org/multierr"
+)
+
+const (
+	requiredErrMsg   = "%q value must be set"
+	outOfRangeErrMsg = "%q is out of range"
+)
+
+var (
+	validatorInstance *v.Validate
+	once              sync.Once
+)
+
+// Get initializes and registers validation tags once, and returns validator instance.
+func Get() *v.Validate {
+	once.Do(func() {
+		validatorInstance = v.New()
+	})
+
+	return validatorInstance
+}
+
+// validate validates structs.
+func validate(s interface{}) error {
```

Comment:
Obviously not something for this PR, but it might be good extracting this into a library. I've seen similar checks in a number of PRs.

Reply:
We will discuss this proposal with the team 👍

Reply:
Additionally, it would be good to have tests for the validator. Personally, I find it OK if we add it separately, if we've tested it indirectly through the config tests.

Reply:
@maha-hajja confirmed this be done in https://github.com/ConduitIO/conduit/issues/680.